src/less/icon/base/icon.less

```diff
@@ -65,6 +65,7 @@ svg {
     }
 
     &.icon--checkbox-checked {
+        .color-token(checkbox-mixed-foreground);
```

Comment:
This should not be needed if the fill is applied on the SVG (I checked it locally)